### PR TITLE
Fix console windows crash

### PR DESCRIPTION
**Pull request by heliclei**

### CODE REVIEW

File: cocos/base/CCConsole.cpp [hunk 1]
```diff
@@ -325,7 +325,12 @@ bool Console::listenOnTCP(int port)
         if (bind(listenfd, res->ai_addr, res->ai_addrlen) == 0)
             break;          /* success */
 
-        close(listenfd);    /* bind error, close and try next one */
+/* bind error, close and try next one */
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32)
+        closesocket(listenfd);
```
Comment (dumganhar) [inline cocos/base/CCConsole.cpp:330]: Should we invoke `WSACleanup` after `closesocket`?

Comment (heliclei) [inline cocos/base/CCConsole.cpp:330]: I think WSACleanup is to clean up the resources alloc by WSAStart(), so, we need only call WSACleanup once when CCConsole exit.

File: tests/test-cpp/Classes/ConsoleTest/ConsoleTest.cpp [hunk 2]
```diff
@@ -165,14 +166,14 @@ void ConsoleTCP::onEnter()
...
-    _console = Director::getInstance()->getConsole();
-
     static struct Console::Command commands[] = {
         {"hello", "This is just a user generated command", [](int fd, const std::string& args) {
+            
             const char msg[] = "how are you?\nArguments passed: ";
-            write(fd, msg, sizeof(msg));
-            write(fd, args.c_str(), args.length());
-            write(fd, "\n",1);
+            auto _console = Director::getInstance()->getConsole();
```
Comment (dumganhar) [inline tests/test-cpp/Classes/ConsoleTest/ConsoleTest.cpp]: This line isn't needed. Right?

Comment (heliclei) [inline tests/test-cpp/Classes/ConsoleTest/ConsoleTest.cpp]: Yes, thanks for pointing this out.

**Comment by dumganhar** [comment]
Thanks. Merged.

**Pull request merged**